### PR TITLE
Add a check for github link to repository for capitalization

### PR DESCRIPTION
**Pull request by lshep**
We had an issue with Laurents package as the github link was all lowercase but the github repo had different capitalization.  Github doesn't care about case so it doesn't matter for a link,  our sqlite database for issues and R code this does matter.   It prevented builds on version bumps.

### CODE REVIEW

File: core.rb [hunk 1]
```diff
@@ -663,6 +674,23 @@ def Core.handle_repo_does_not_exist(repos_url, issue_number, login, close=true)
...
+
+      The github link provided above https://github.com/#{repos_url} ,
+      does not match the capitalization of the github repository.
+      Please update.
+    END
+    if close
+      comment += "I am closing this issue. Please try again with a new issue."
+      Core.close_issue(issue_number)
+    end
+    Octokit.add_comment(Core::NEW_ISSUE_REPO, issue_number, comment)
```
Comment (mtmorgan) [inline core.rb]: Should the order of close / add_comment be reversed here, so that the comment is added and then the issue closed?

File: core.rb [hunk 2]
```diff
@@ -632,6 +637,12 @@ def Core.repo_exists_in_github? (repos_url)
     end
   end
 
+  def Core.repo_caps_check? (repos_url, full_repos_url)
```
Comment (mtmorgan) [inline core.rb]: To be sure I understand, we already check that the repository and the package have the same name / capitalization, and now we are making sure that the repository and URL have the same capitalization? Does it make better sense to use the DESCRIPTION file as definitive for the package name, i.e., doing whatever is done to parse the DESCRIPTION Package: field for the check between repository & package?

File: core.rb [hunk 3]
```diff
@@ -9,6 +9,7 @@
 require 'aws-sdk'
 require 'stomp'
 require 'open-uri'
+require 'nokogiri'
```
Comment (mtmorgan) [inline core.rb:12]: I guess this introduces a (pretty heavy) dependency? and so should be avoided if possible; if the comment below about checking the package name from the DESCRIPTION file is appropriate, then this dependency could be removed

File: core.rb (hunk 1 above)
Comment (lshep) [reply core.rb]: I modelled it after the handle_repo_does_not_exist  code that was already present but yes the order should probably change.

File: core.rb (hunk 2 above)
Comment (lshep) [reply core.rb]: The issue was in the sqlite database that has the github link to map the issue -  so this is done before we even get to checking or looking at the package

Comment (lshep) [reply core.rb]: so on issues.bioconductor.org in the sqlite database -   it gets filled -  and the name field needed to match the github repository capitalization
```
id|name|pw_hash|issue_number|login
1|dtenenba/spbtest|$2a$10$X1QM9XUSj94zxjtbXbtEpeQ4gAFY5NLCDLkgMjA6sOkkCtchJM5p6|4|dtenenba
```

File: core.rb (hunk 3 above)
Comment (lshep) [reply core.rb:12]: if there is another way to read in the file and parse it I'm open to suggestions.   The database still needs the correct link

Comment (lshep) [reply core.rb:12]: The organization capitalization also has to be correct  (ie.  Bioconductor/spbtest3  couldn't be   bioconductor/sbptest3)   and the organization of hte github account isnt in the description

Comment (lshep) [reply core.rb:12]: So the description file can't be used

Comment (mtmorgan) [reply core.rb:12]: I didn't realize that the 'organization' (or sometimes individual, i guess) part of the URL needed to match; would it be possible to be more explicit in the error message, e.g., the "Capitalization in the url provided above {{}} must match capitalization of the repository url {{}}" or if substitution is difficult then be explicit about what needs to match with a text description?

Comment (lshep) [reply core.rb:12]: I'll look more into seeing if there is a better dependency that could parse the html page -  or maybe at least a lighter one - 

Yes the more specific message should be doable - I'll just have to change the helper functions a little bit and possible add a different one...

Comment (lshep) [reply core.rb]: nokogiri is what mostly comes up in searches and is the top recommended on the ruby-toolbox site ... https://www.ruby-toolbox.com/categories/html_parsing.html

**Comment by lshep** [comment]
I changed the helper function to return the actual organization/repo name and then do the check manually so that if it fails it can be included in the message.  

nokogiri is what mostly comes up in searches for html parsing and is the top recommended on the ruby-toolbox site ... https://www.ruby-toolbox.com/categories/html_parsing.html  - if you would like me to invesitage one of the other listed on the page please let me know which one.

**Review by mtmorgan — APPROVED** [review]

**Comment by lshep** [comment]
@mtmorgan   in case this causes unforseen issues I'll plan on merging and pushing up next week monday then?

**Comment by mtmorgan** [comment]
yep next week sounds fine thanks!

**Pull request merged**